src/thriveopedia/fossilisation/FossilisedSpecies.cs

```diff
@@ -161,7 +180,6 @@ private static void WriteDataToFossilFile(string target, string justInfo, string
         using var file = new File();
         if (file.Open(target, File.ModeFlags.Write) != Error.Ok)
         {
-            GD.PrintErr("Cannot open file for writing: ", target);
             throw new IOException("Cannot open: " + target);
```

Comment:
For here too (although you didn't write that).

Reply:
Just this one place should not be refactored. The saving system shows how it is done: you first check that the file can be written (and show an error dialog if there is a problem), then write the file (which can fail with an exception). Or at least that's how I remember.

Reply:
I'm looking through the save stuff a little. It seems to me that the methods in Save.cs throw exceptions, but SaveHelper.cs catches those exceptions and then handles them more elegantly. If that sounds right, I'm thinking FossilisedSpecies.cs should throw exceptions still, but ThriveopediaMuseumPage.cs should then catch those and display a popup instead. Let me know if I've missed something important

Reply:
I think that makes sense (though having written the saving code I might be a bit biased). It makes sense to me that the low level operation communicates with exceptions which the higher level operation can then convert into a GUI for notifying the user about the problem.

Reply:
The other way I would think to do it would be through returning error codes, which I personally like a lot less.

Reply:
Returning error codes should be avoided. That's a pattern that is not used in the codebase.